data/bookkeeping/block.go

```diff
@@ -539,6 +560,31 @@ func (block Block) paysetCommit(t config.PaysetCommitType) (crypto.Digest, error
 	}
 }
 
+func (block Block) paysetCommitSHA256() (crypto.Digest, error) {
+	params, ok := config.Consensus[block.CurrentProtocol]
+	if !ok {
+		return crypto.Digest{}, fmt.Errorf("unsupported protocol %v", block.CurrentProtocol)
+	}
+
+	if !params.EnableSHA256TxnRootHeader {
+		return crypto.Digest{}, nil
+	}
+
+	tree, err := block.TxnMerkleTreeSHA256()
+	if err != nil {
+		return crypto.Digest{}, err
+	}
+	// in case there are no leaves (e.g empty block with 0 txns) the merkle root is a slice with length of 0.
```

Comment:
Since we use VC. the root will never be empty (even if there are no txn in the blocks).
This is due to the fact that we pad the VC to a full tree.